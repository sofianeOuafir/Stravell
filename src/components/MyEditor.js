import React from "react";
import Editor from "draft-js-plugins-editor";
import createUndoPlugin from "draft-js-undo-plugin";
import createEmojiPlugin from 'draft-js-emoji-plugin';

const emojiPlugin = createEmojiPlugin();
const { EmojiSuggestions, EmojiSelect } = emojiPlugin;
const undoPlugin = createUndoPlugin();
const { UndoButton, RedoButton } = undoPlugin;

class MyEditor extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
        <Editor plugins={[undoPlugin, emojiPlugin]} {...this.props} />
        {!this.props.readOnly && (
          <div>
            <UndoButton />
            <RedoButton />
            <EmojiSuggestions />
            <EmojiSelect />
          </div>
        )}
      </div>
    );
  }
}

export default MyEditor;
