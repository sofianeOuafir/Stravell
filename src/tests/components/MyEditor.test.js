import React from 'react';
import { shallow } from 'enzyme';
import { EditorState } from "draft-js";
import MyEditor from './../../components/MyEditor';


const editorState = EditorState.createEmpty();
let props = { editorState, onChange: () => {} };

describe('editor is read only', () => {
  test('should render correctly', () => {
    props.readOnly = true;
    const wrapper = shallow(<MyEditor { ...props } />)
    expect(wrapper).toMatchSnapshot()
  });
});

describe('editor is not read only', () => {
  test('should render correctly', () => {
    props.readOnly = false;
    const wrapper = shallow(<MyEditor { ...props } />)
    expect(wrapper).toMatchSnapshot()
  });
});