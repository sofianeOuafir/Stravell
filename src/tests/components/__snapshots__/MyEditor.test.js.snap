// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editor is not read only should render correctly 1`] = `
<div
  className="editor-container"
  onClick={[Function]}
>
  <input
    className="hide"
    id="imageUploads"
    onChange={[Function]}
    type="file"
  />
  <PluginEditor
    customStyleMap={Object {}}
    decorators={Array []}
    defaultBlockRenderMap={true}
    defaultKeyBindings={true}
    defaultKeyCommands={true}
    editorState={Object {}}
    onChange={[Function]}
    plugins={
      Array [
        Object {
          "SideToolbar": [Function],
          "initialize": [Function],
          "onChange": [Function],
        },
        Object {
          "decorators": Array [
            Object {
              "component": [Function],
              "strategy": [Function],
            },
          ],
        },
        Object {
          "LinkButton": [Function],
          "decorators": Array [
            Object {
              "component": [Function],
              "matchesEntityType": [Function],
              "strategy": [Function],
            },
          ],
          "initialize": [Function],
        },
        Object {
          "handleDroppedFiles": [Function],
        },
        Object {
          "RedoButton": [Function],
          "UndoButton": [Function],
          "initialize": [Function],
        },
        Object {
          "addImage": [Function],
          "blockRendererFn": [Function],
        },
        Object {
          "AlignmentTool": [Function],
          "blockRendererFn": [Function],
          "decorator": [Function],
          "initialize": [Function],
        },
        Object {
          "blockRendererFn": [Function],
          "decorator": [Function],
          "handleKeyCommand": [Function],
          "handleReturn": [Function],
          "keyBindingFn": [Function],
          "onChange": [Function],
          "onDownArrow": [Function],
          "onUpArrow": [Function],
        },
        Object {
          "blockRendererFn": [Function],
          "decorator": [Function],
          "initialize": [Function],
        },
        Object {
          "decorator": [Function],
          "handleDrop": [Function],
          "initialize": [Function],
        },
      ]
    }
    readOnly={false}
  />
  <div>
    <Decorated(AlignmentTool) />
    <Decorated(Toolbar) />
  </div>
</div>
`;

exports[`editor is read only should render correctly 1`] = `
<div
  className={false}
  onClick={[Function]}
>
  <input
    className="hide"
    id="imageUploads"
    onChange={[Function]}
    type="file"
  />
  <PluginEditor
    customStyleMap={Object {}}
    decorators={Array []}
    defaultBlockRenderMap={true}
    defaultKeyBindings={true}
    defaultKeyCommands={true}
    editorState={Object {}}
    onChange={[Function]}
    plugins={
      Array [
        Object {
          "SideToolbar": [Function],
          "initialize": [Function],
          "onChange": [Function],
        },
        Object {
          "decorators": Array [
            Object {
              "component": [Function],
              "strategy": [Function],
            },
          ],
        },
        Object {
          "LinkButton": [Function],
          "decorators": Array [
            Object {
              "component": [Function],
              "matchesEntityType": [Function],
              "strategy": [Function],
            },
          ],
          "initialize": [Function],
        },
        Object {
          "handleDroppedFiles": [Function],
        },
        Object {
          "RedoButton": [Function],
          "UndoButton": [Function],
          "initialize": [Function],
        },
        Object {
          "addImage": [Function],
          "blockRendererFn": [Function],
        },
        Object {
          "AlignmentTool": [Function],
          "blockRendererFn": [Function],
          "decorator": [Function],
          "initialize": [Function],
        },
        Object {
          "blockRendererFn": [Function],
          "decorator": [Function],
          "handleKeyCommand": [Function],
          "handleReturn": [Function],
          "keyBindingFn": [Function],
          "onChange": [Function],
          "onDownArrow": [Function],
          "onUpArrow": [Function],
        },
        Object {
          "blockRendererFn": [Function],
          "decorator": [Function],
          "initialize": [Function],
        },
        Object {
          "decorator": [Function],
          "handleDrop": [Function],
          "initialize": [Function],
        },
      ]
    }
    readOnly={true}
  />
</div>
`;
